python/plugins/processing/tests/QgisAlgorithmsTest.py

```diff
@@ -108,6 +116,57 @@ def testParameterPythonImport(self):
             # and now we should be able to instantiate an object!
             exec('test = {}(\'id\',\'name\')\nself.assertIsNotNone(test)'.format(t.className()))
 
+    def testCreateGridFromLayout(self):
+
+        testDataPath = os.path.join(os.path.dirname(__file__), 'testdata')
+        temp_dir = tempfile.mkdtemp()
+
+        p = QgsProject().instance()
+        source = os.path.join(testDataPath, 'airports.gml')
+        vl = QgsVectorLayer(source)
+        self.assertTrue(vl.isValid())
+        p.addMapLayer(vl)
+
+        pl = QgsPrintLayout(p)
+
+        doc = QDomDocument()
+        f = open(os.path.join(testDataPath, 'test_layout.qpt'))
+        doc.setContent(f.read())
+        f.close()
+
+        pl.loadFromTemplate(doc, QgsReadWriteContext())
+        layoutname = pl.name()
+        mapuuid = pl.referenceMap().uuid()
+
+        lomanager = p.layoutManager()
+        lomanager.addLayout(pl)
+
+        alg = QgsApplication.processingRegistry().createAlgorithmById('qgis:createatlasgrid')
+
+        self.assertIsNotNone(alg)
+
+        temp_file = os.path.join(temp_dir, 'grid.shp')
+        parameters = {'LAYOUT': layoutname,
+                      'MAP': mapuuid,
+                      'COVERAGE_LAYER': vl,
+                      'COVERAGE_BUFFER': 0,
+                      'ONLY_ON_FEATURES': True,
+                      'OUTPUT': temp_file}
+        context = QgsProcessingContext()
+        context.setProject(p)
+        feedback = QgsProcessingFeedback()
+
+        results, ok = alg.run(parameters, context, feedback)
+
+        self.assertTrue(ok)
+        self.assertTrue(os.path.exists(temp_file))
+
+        res = QgsVectorLayer(temp_file, 'res')
+        self.assertTrue(res.isValid())
+        self.assertEqual(res.featureCount(), 37)
```

Comment:
can you also check the results here? e.g. scan through for an some expected geometries (as wkt)